File: easy/3536-maximum-product-of-two-digits/solution.py
class Solution:
    def maxProduct(self, n: int) -> int:
        """
        3536. Maximum Product of Two Digits
        Time: O(d)
        Space: O(1)
        """
        max1 = max2 = -1  # two largest digit values
        for ch in str(n):
            d = int(ch)
            if d >= max1:
                max2 = max1
                max1 = d
            elif d > max2:
                max2 = d
        return max1 * max2

File: easy/3536-maximum-product-of-two-digits/solution.java
/**
 * 3536. Maximum Product of Two Digits
 * Time: O(d)
 * Space: O(1)
 */
class Solution {
    public int maxProduct(int n) {
        int max1 = -1, max2 = -1; // two largest digit values
        while (n > 0) {
            int d = n % 10;
            if (d >= max1) {
                max2 = max1;
                max1 = d;
            } else if (d > max2) {
                max2 = d;
            }
            n /= 10;
        }
        return max1 * max2;
    }
}
